File: Array/Plus One.py
class Solution(object):
    
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        
        i = -1
        
        while (i + len(digits) > -1):
            if digits[i] == 9:
                digits[i] = 0
            else:
                digits[i] += 1
                return digits
            i -= 1
        
        if digits[0] == 0:
            digits.insert(0, 1)
            return digits

File: Plus One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        bool plus = true;
        int i = digits.size() - 1;
        while ((i >= 0) and plus)
        {
            digits[i] = (digits[i] + 1) % 10;
            plus = (digits[i] == 0);
            i -= 1;
        }
        if (digits[0] == 0)
        {
            digits.push_back(0);
            digits[0] = 1;
        }
        return digits;
    }
};
